kernel/tests/write.rs

```diff
@@ -251,3 +270,377 @@ async fn test_invalid_commit_info() -> Result<(), Box<dyn std::error::Error>> {
     ));
     Ok(())
 }
+
+// check that the timestamps in commit_info and add actions are within 10s of SystemTime::now()
+fn check_action_timestamps<'a>(
+    parsed_commits: impl Iterator<Item = &'a serde_json::Value>,
+) -> Result<(), Box<dyn std::error::Error>> {
+    // check that timestamps are within 10s of SystemTime::now()
+    let now: i64 = std::time::SystemTime::now()
+        .duration_since(std::time::UNIX_EPOCH)?
+        .as_millis()
+        .try_into()
+        .unwrap();
+
+    parsed_commits.for_each(|commit| {
+        if let Some(commit_info_ts) = &commit.pointer("/commitInfo/timestamp") {
+            assert!(now - commit_info_ts.as_i64().unwrap() < 10_000);
+        }
+        if let Some(add_ts) = &commit.pointer("/add/modificationTime") {
+            assert!(now - add_ts.as_i64().unwrap() < 10_000);
+        }
+    });
+
+    Ok(())
+}
+
+// update `value` at (.-separated) `path` to `new_value`
+fn set_value(
+    value: &mut serde_json::Value,
+    path: &str,
+    new_value: serde_json::Value,
+) -> Result<(), Box<dyn std::error::Error>> {
+    let mut path_string = path.replace(".", "/");
```

Comment:
curious, why not just take a json pointer string as the path?

Reply:
eh just used to the `field1.field2` syntax so was trying to make it easier on future usage (instead of having to write `/field1/field2`

happy to change if the latter seems better